File: ops/frontend/src/types/account.ts
import type { Status, TimeSeriesPoint } from './common'

export type AccountType = 'live' | 'demo' | 'prop'

/** A trading account held at a broker. */
export interface Account {
  id: string
  label: string
  broker: string
  type: AccountType
  currency: string
  status: Status
  balance: number
  equity: number
  /** Realised pnl booked today. */
  todayPnl: number
  /** Unrealised pnl on open positions. */
  openPnl: number
  /** Margin level (equity / margin) as a percentage. */
  marginLevelPct: number
  leverage: number
  openPositions: number
  /** Strategy codes deployed against this account. */
  strategies: string[]
  /** Equity history for the account sparkline. */
  equityCurve: TimeSeriesPoint[]
}

File: ops/backend/app/schemas/account.py
"""Account schemas."""

from typing import Literal

from pydantic import BaseModel

from app.schemas.common import Status, TimeSeriesPoint

AccountType = Literal["live", "demo", "prop"]


class Account(BaseModel):
    """A trading account held at a broker."""

    id: str
    label: str
    broker: str
    type: AccountType
    currency: str
    status: Status
    balance: float | None = None
    equity: float | None = None
    todayPnl: float | None = None
    openPnl: float | None = None
    marginLevelPct: float | None = None
    leverage: int | None = None
    openPositions: int | None = None
    strategies: list[str]
    equityCurve: list[TimeSeriesPoint]
